Framework: Move currentUser Redux state to own root key

aduth opened the pull request:
This pull request seeks to move current user state from `state.ui.currentUserId` to `state.currentUser.id`, reflecting this change in the folder structure. Despite the number of changes, no functions have changed, nor have any been added or removed.

__Testing instructions:__

Current user is not used by any area of the application, though there are a few pull requests open making  use of it that will need to be updated to use `getCurrentUser` from `state/current-user/selectors` instead of `state/ui/selectors` (#2322, #2472).

Confirm that Mocha tests pass for both current user and UI state by running `make test` from the project root directory or directly from `client/state/current-user` and `client/state/ui`.

/cc @mtias

jordwest commented:
The tests pass for me, however there's one usage of `setCurrentUserId` that's preventing boot in [`client/boot/index.js on line 26`](https://github.com/Automattic/wp-calypso/blob/update/redux-current-user/client/boot/index.js#L26)

aduth commented:
>The tests pass for me, however there's one usage of `setCurrentUserId` that's preventing boot in [`client/boot/index.js on line 26`](https://github.com/Automattic/wp-calypso/blob/update/redux-current-user/client/boot/index.js#L26).

Ouch, I'm sure I would have caught that in my final testing, though a bit embarassing to have left out in the original changes. Rebased and updated to include the missing change.

aduth commented:
Rebased #2322 and #2472 to account for the changes here. Searched and could not find other pull requests that would be in a similar situation.
